### PR TITLE
Integrate atom tools and other goodies

### PR DESCRIPTION
**Pull request by prabhu**
Container image for testing.

```
docker pull ghcr.io/appthreat/atom:feature-atom-tools
```

Example for a java project:

```shell
cd app directory
docker run --rm -v /tmp:/tmp -v $HOME:$HOME -v $(pwd):/app:rw -it ghcr.io/appthreat/atom:feature-atom-tools atom usages --extract-endpoints -o /app/app.atom --slice-outfile /app/usages.slices.json -l java /app
```

### CODE REVIEW

File: src/main/scala/io/appthreat/atom/Atom.scala [hunk 1]
```diff
@@ -351,8 +361,20 @@ object Atom:
                             AtomDataFlowSlice(x, DataFlowGraph.buildFromSlice(x).paths).toJson
                         )
                     saveSlice(config.outputSliceFile, atomDataFlowSliceJson)
-                case _: UsagesConfig =>
+                case u: UsagesConfig =>
                     saveSlice(config.outputSliceFile, sliceCpg(ag).map(_.toJson))
+                    if u.extractEndpoints then
+                        val result = ExternalCommand.run(
+                          s"atom-tools convert -i ${config.outputSliceFile} -t ${config.language} -f openapi3.1.0 -q -o ${config.inputPath.pathAsString}${java.io.File.separator}openapi.generated.json",
```
Comment (prabhu) [inline src/main/scala/io/appthreat/atom/Atom.scala:368]: This is the place where atom-tools is invoked.

Comment (cerrussell) [reply src/main/scala/io/appthreat/atom/Atom.scala:368]: @prabhu Cool! I think I had still better create an atom-tools container and probably binaries though as fwdsec wish to utilize filtering - unless you want to allow that here.

**Review by cerrussell — APPROVED** [review]

File: src/main/scala/io/appthreat/atom/Atom.scala (hunk 1 above)
Comment (prabhu) [reply src/main/scala/io/appthreat/atom/Atom.scala:368]: @cerrussell changed the dockerfile to use CMD so atom-tools can be invoked directly from atom container image too

File: src/main/scala/io/appthreat/atom/Atom.scala [hunk 2]
```diff
@@ -425,130 +448,128 @@ object Atom:
...
+                        )
+                        .map { ag =>
+                            new JavaScriptInheritanceNamePass(ag).createAndApply()
+                            new ConstClosurePass(ag).createAndApply()
+                            new ImportResolverPass(ag).createAndApply()
+                            new JavaScriptTypeRecoveryPass(ag).createAndApply()
+                            new TypeHintPass(ag).createAndApply()
+                            ag
+                        }
+                case Languages.PYTHONSRC | Languages.PYTHON | "PY" =>
```
Comment (buchi-busireddy) [inline src/main/scala/io/appthreat/atom/Atom.scala:529]: @prabhu I'm just curious: What's the exact change here? This code seems precisely the same as the previous code, right?

File: src/main/scala/io/appthreat/atom/slicing/UsageSlicing.scala [hunk 3]
```diff
@@ -246,7 +246,17 @@ object UsageSlicing:
                         m.columnNumber.map(_.intValue())
                       )
                   )
-                  .l,
+                  .l ++ call.argument.inCall.map(c =>
```
Comment (prabhu) [inline src/main/scala/io/appthreat/atom/slicing/UsageSlicing.scala:249]: @buchi-busireddy This append operation results in new entries in the usages slices that captures the code and the types suitable for languages like python

**Pull request merged**